expfmt/text_parse.go

```diff
@@ -299,6 +299,25 @@ func (p *TextParser) startLabelName() stateFn {
 		p.parseError(fmt.Sprintf("expected '=' after label name, found %q", p.currentByte))
 		return nil
 	}
+	// Check for duplicate label names.
+	if len(p.currentMetric.Label) > 1 {
```

Comment:
You don't need this check.

Reply:
I was trying to prevent the code below from executing if there is only one label. But I can remove it.

Reply:
The for loop already handles that correctly.